projects/github-actions/repo-gardening/src/utils/labels/get-all-labels.js

```diff
@@ -0,0 +1,43 @@
+/* global GitHub */
+const debug = require( '../debug' );
+
+// Cache for getLabels.
+const cache = {};
+
+/**
+ * Get all the labels available in the repo.
+ *
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {string} owner   - Repository owner.
+ * @param {string} repo    - Repository name.
+ * @param {string} filter  - Optionally filter to only return a subset of labels. Use a regex pattern.
+ * @return {Promise<Array>} Promise resolving to an array of all labels in the repo.
+ */
+async function getAllLabels( octokit, owner, repo, filter = '' ) {
+	const labelList = [];
+	const cacheKey = `${ owner }/${ repo } #${ filter }`;
+	if ( cache[ cacheKey ] ) {
+		debug( `get-all-labels: Returning list of labels for ${ cacheKey } from cache.` );
+		return cache[ cacheKey ];
+	}
+
+	debug( `get-all-labels: Get list of labels for ${ cacheKey }.` );
+
+	for await ( const response of octokit.paginate.iterator( octokit.rest.issues.listLabelsForRepo, {
+		owner,
+		repo,
+		per_page: 100,
+	} ) ) {
+		for ( const label of response.data ) {
+			if ( filter && ! label.name.match( filter ) ) {
+				continue;
+			}
+			labelList.push( label );
+		}
+	}
+
+	cache[ cacheKey ] = labelList;
+	return labelList;
```

Comment:
Probably more efficient to do it like this. The API call isn't changing based on the `filter` parameter.
```suggestion
	let labelList;
	const cacheKey = `${ owner }/${ repo }`;
	if ( cache[ cacheKey ] ) {
		debug( `get-all-labels: Using list of labels for ${ cacheKey } from cache.` );
		labelList = cache[ cacheKey ];
	} else {
		debug( `get-all-labels: Get list of labels for ${ cacheKey }.` );

		labelList = [];
		for await ( const response of octokit.paginate.iterator( octokit.rest.issues.listLabelsForRepo, {
			owner,
			repo,
			per_page: 100,
		} ) ) {
			for ( const label of response.data ) {
				labelList.push( label );
			}
		}
		cache[ cacheKey ] = labelList;
	}

	return filter ? labelList.filter( label => label.name.match( filter ) : labelList;
```

Or you might even drop the `filter` parameter entirely, and let the caller do `.filter( whatever )` if they want to filter the list.

Reply:
done in f42ae3ac07652cd8d4ca06be8680729e6883789c

> Or you might even drop the filter parameter entirely, and let the caller do .filter( whatever ) if they want to filter the list.

I was thinking about that too, but I figure we may need it for other things in the future.